ros_buildfarm/config/doc_build_file.py

```diff
@@ -72,9 +74,16 @@ def __init__(self, name, data):
             self.doc_repositories = data['doc_repositories']
             assert isinstance(self.doc_repositories, list)
 
-        # doc_repositories can only be used with doc type make_target
+        # doc_repositories can only be used with make_target and external_site
+        # doc types
         is_make_target_type = self.documentation_type == DOC_TYPE_MAKE
-        assert not self.doc_repositories or is_make_target_type
+        is_external_site_type = self.documentation_type == DOC_TYPE_EXTERNAL
+        are_doc_repositories_allowed = (is_make_target_type or
+                                        is_external_site_type)
+        assert not self.doc_repositories or are_doc_repositories_allowed
```

Comment:
With the logic getting more complex I would suggest to decouple this like:

```
if self.documentation_type not in (DOC_TYPE_MAKE, DOC_TYPE_EXTERNAL):
    assert not self.doc_repositories
```

Reply:
Good point, done.